src/v2i-hub/CDASimAdapter/src/CDASimAdapter.cpp

```diff
@@ -161,6 +168,34 @@ namespace CDASimAdapter{
         }
     }
 
+    void CDASimAdapter::start_sensor_detected_object_detection_thread() {
+        PLOG(logDEBUG) << "Creating Thread Timer for simulated external object" << std::endl;
+        try 
+        {
+            if(!external_bject_detection_thread_timer)
+            {
+                external_bject_detection_thread_timer  = std::make_unique<tmx::utils::ThreadTimer>();
+            }            
+            external_bject_detection_thread_timer->AddPeriodicTick([this](){
+                PLOG(logDEBUG1) << "Listening for Sensor Detected Message from CDASim." << std::endl;
+                auto msg = connection->consume_sensor_detected_object_message();
+                if ( !msg.is_empty()) {             
+                    this->forward_simulated_detected_message(msg);
+                }
+                else 
+                {
+                    PLOG(logDEBUG1) << "CDASim connection has not yet received an simulated sensor detected message!" << std::endl;
+                }
+            }//End lambda
+            , std::chrono::milliseconds(100));
+            external_bject_detection_thread_timer->Start();
```

Comment:
looks like there may be a type in this variable name. Not a big deal,just in case you didnt notice